mode-apis-base/src/entityModels.ts

```diff
@@ -442,7 +469,15 @@ export interface FetchEntityFilters {
     // An comma separated list of entity classes
     readonly entityClasses?: string | undefined;
     readonly parentId?: string | null | undefined;
+    // To tell the backend whether or not to return only the DIRECT children or includes all descendants.
+    // By default, the back end will only include the entity instances that are direct children of the specified "parentId". If this flag
+    // is set to true, the back end will also includes children, grandchildren, great grandchildren, etc... of the "parentId"
+    readonly recursive?: boolean | undefined;
     readonly attributesQuery?: string | undefined;
+
+    // To have the back end sort the results by the specified "sortBy" (attribute name) in the specified "sortOrder"
+    readonly sortBy?: string;
+    readonly sortOrder?: SortOrder;
```

Comment:
NOTE: Ignore the `sortOrder`, this is for front end use only. To make things simple, the front end will separate these 2 fields, `sortBy` and `sortOrder`. Before making the API call, the front end will concatenate them and will only pass `sortBy` to the back end.

Reply:
I think we can optimize the backend implementation in the same way as the frontend at this point. (cc @jamesbalcombe83 )

Reply:
We can keep it consistent with the other APIs which use `sortBy` only. We don't need to change the back end for this because if we implement this endpoint to allow sortBy and sortOrder, we might need to change the other endpoints as well.

The only issue with using just the sortBy param, we have to use `:` separator. This means the sort by field name cannot have `:` in the name. If we are sure attribute names can't have `:` in the names then we are good.

Reply:
attribute keys must meet [these rules](https://github.com/moderepo/platform/blob/9fa324ba560712788e214e75f515627c53f97b84/components/pkg/entity_helper/entity_class_validators.go#L22) I think that precludes `:`

Reply:
which MODE api currently uses `sortBy`?

Reply:
The GET home videos API `/homes/${homeId}/smartModules/${videoSmartModuleId}/videos`

Reply:
https://dev.tinkermode.com/platform/api/video

I'm regretting this design😇

Reply:
It seems bad 😢

Reply:
So, I don't think we must follow this "existing" implementation as our convention. (hopefully, we can change the video implementation later for the sake of consistency)

Reply:
(to be fair, several public APIs support the sortBy parameter in the same manner as the video API.

Reply:
yeah, I am ok either way. I think we only have 2 APIs that accept `sortBy` param that way so it is not too hard to change them later.

Reply:
> If we are sure attribute names can't have : in the names then we are good.

By the way, it's okay to search the field to have `:`. It's distinguishable by the regex; `s/.+:(asc|desc)$/`.

Reply:
Let's be consistent with the current implementations `sortBy` is actually `sortField:sortOrder`